src/validation.cpp

```diff
@@ -2445,6 +2445,7 @@ bool CChainState::ConnectTip(CValidationState& state, const CChainParams& chainp
             return error("%s: ConnectBlock %s failed, %s", __func__, pindexNew->GetBlockHash().ToString(), FormatStateMessage(state));
         }
         nTime3 = GetTimeMicros(); nTimeConnectTotal += nTime3 - nTime2;
+        assert(nBlocksTotal > 0);
```

Comment:
I'd avoid this, I think it is noise as there is already a "divide by nBlocksTotal" which implies this assertion.

Reply:
Not a fan of silently committing UB.

Reply:
There are 11 occurrences of `/ nBlocksTotal`.